_book/02-first-steps/05-npm-usage.md

```diff
@@ -0,0 +1,74 @@
+---
+title: "Usage with npm"
+---
+
+The latest stable version is available via NPM registry. The version distributed via NPM it the WebAssembly build. It can be installed with: 
+
+```bash
+npm install verovio
+```
+
+## Basic usage
+
+```javascript
+const verovio = require('verovio');
+const fs = require('fs');
+
+/* Wait for verovio to load */
+verovio.module.onRuntimeInitialized = function ()
+{
+   // create the toolkit instance
+   const vrvToolkit = new verovio.toolkit();
+   // read the MEI file
+	mei = fs.readFileSync('hello.mei');
+   // load the MEI data as string into the toolkit
+	vrvToolkit.loadData(mei.toString());
+   // render the fist page as SVG
+	svg = vrvToolkit.renderToSVG(1, {});
+   // save the SVG into a file
+	fs.writeFileSync('hello.svg', svg);
+}
+```
+
+
+## Usage with ESM
+
+Since version 3.11.0 there is an ESM compatible version of the npm package with a modularized build of the Verovio module, that needs a slightly different setup to wait for the asynchronous Emscripten module to be ready for usage which is now [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) based instead of using the `onRuntimeInitialized` callback function.
+
+ Use `.mjs` as file extension when using this directly in Node.js or set `"type": "module"` in your `package.json`.
```

Comment:
I'm not sure I really understand what this means... where would I use `.mjs`? It's not in any of the examples...

Reply:
I hope my comment from before will explain this. You can force ESM usage in Node.js if you use `.mjs` as a file extension. Then you can use the modern import syntax. Also have a loook here: https://github.com/WolfgangDrescher/verovio-npm-examples/tree/master/node-esm

Reply:
Yes, it helps *me*, but since this is going into the book it will need to help others too, and they probably will not know about your comment. :-D

Reply:
@WolfgangDrescher please integrate the explanations you gave here as comments into the book. Thanks